tools/pepsi/src/upload.rs

```diff
@@ -93,7 +95,14 @@ pub async fn upload(arguments: Arguments, repository: &Repository) -> Result<()>
         let hulk_directory = hulk_directory.clone();
         async move {
             let nao = Nao::new(nao_address.ip);
+
+            if !arguments.skip_has_stable_os_version {
+                if !nao.has_stable_os_version().await {
+                    return Ok(());
+                }
+            }
             println!("Starting upload to {nao_address}");
+
```

Comment:
If you insert empty lines, you are grouping statements. Always think about how you are grouping them and if they belong together as this structures your code. Here, you changed the grouping but I would argue that the "Starting upload" message does not belong to your version check but rather to the upload process afterwards.